LLama/Native/LLamaTokenDataArray.cs

```diff
@@ -342,13 +362,28 @@ public struct LLamaTokenDataArrayNative
         /// A pointer to an array of LlamaTokenData
         /// </summary>
         /// <remarks>Memory must be pinned in place for all the time this LLamaTokenDataArrayNative is in use</remarks>
-        public IntPtr data;
+        private unsafe LLamaTokenData* _data;
 
         /// <summary>
         /// Number of LLamaTokenData in the array
         /// </summary>
         public ulong size;
-
+        
+        /// <summary>
+        /// A pointer to an array of LlamaTokenData
+        /// </summary>
+        /// <remarks>Memory must be pinned in place for all the time this LLamaTokenDataArrayNative is in use</remarks>
+        public Span<LLamaTokenData> data
```

Comment:
I like that this is exposed! However what do you mean in the comment? Is it correct?
Might need some clarification if intended for end-users.

Reply:
I've removed that comment, it's not really relevant any more.

When you could pass in a pointer directly, it was your responsibility to ensure that the pointer was fixed (i.e. `Memory.Pin()` or `fixed (blah)`) for all the time that memory was in use.

However, now that `Create` is the only way a `NativeTokenDataArray` can be created that's no longer relevant - `Create` handles it for you by pinning the memory and returning the handle.